File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> lookup;
        for(int i=0; i<nums.size();i++){
            int k=target-nums[i];
            auto it=lookup.find(k);
            if(lookup.find(k)!=lookup.end())
                return {it->second,i};
            else
                lookup.insert({nums[i],i});
        }
    }
};

File: Data Structures & Algorithms/search-insert-position/submission-1.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int l=0;
        int m = nums.length-1;
        int mid=0;
        while(l<=m){
            mid = l+(m-l)/2;
            if(nums[mid]>target){
                m=mid-1;
            }
            else if(nums[mid]<target){
                l=mid+1;
            }
            else{
                return mid;
            }
        }
        return l;
    }
}